src/test/java/io/fusionauth/samlv2/service/DefaultSAMLv2ServiceTest.java

```diff
@@ -1004,13 +1024,47 @@ public void roundTripAuthnRequest_ECDSA(Binding binding) throws Exception {
       request = service.parseRequestPostBinding(queryString, authRequest -> new TestPostBindingSignatureHelper(KeySelector.singletonKeySelector(kp.getPublic()), true));
     }
 
-    // Assert the the parsed request
+    // Assert the parsed request
     assertEquals(request.id, "foobarbaz");
     assertEquals(request.issuer, "https://local.fusionauth.io");
     assertEquals(request.nameIdFormat, NameIDFormat.EmailAddress.toSAMLFormat());
     assertEquals(request.version, "2.0");
   }
 
+  @Test(dataProvider = "assertionEncryption")
+  public void roundTripResponseEncryptedAssertion(EncryptionAlgorithm encryptionAlgorithm, KeyLocation keyLocation,
+                                                  KeyTransportAlgorithm transportAlgorithm, DigestAlgorithm digest,
+                                                  MaskGenerationFunction mgf)
+      throws Exception {
+    KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+    kpg.initialize(2048);
+    KeyPair signingKeyPair = kpg.generateKeyPair();
+    KeyPair encryptionKeyPair = kpg.generateKeyPair();
+
+    byte[] ba = Files.readAllBytes(Paths.get("src/test/xml/encodedResponse.txt"));
+    String encodedResponse = new String(ba);
+    DefaultSAMLv2Service service = new DefaultSAMLv2Service();
+    AuthenticationResponse response = service.parseResponse(encodedResponse, false, null);
+
+    String encodedXML = service.buildAuthnResponse(
```

Comment:
Support for SAML assertion decryption has not been added yet. This test will be expanded to decrypt the encrypted assertion.

For now, this test confirms that we are able to successfully encrypt the assertion with a variety of different encryption parameters.